internal/lib/container_server.go

```diff
@@ -102,6 +104,24 @@ func New(ctx context.Context, configIface libconfig.Iface) (*ContainerServer, er
 		return nil, fmt.Errorf("cannot create container server: interface is nil")
 	}
 
+	if ShutdownWasUnclean(config) {
+		checkOptions := cstorage.CheckEverything()
+		report, err := store.Check(checkOptions)
+		if err != nil {
+			return nil, err
```

Comment:
@nalind I would say, ideally, cri-o does not exit in this case. What cases would cause this to fail? do you think it makes sense to just remove the c/storage like we did before in this error case?

Reply:
Generally, these errors are going to be about permissions or locking.  If returning here means we go from automatically wiping storage to requiring manual intervention, that seems excessive.

Reply:
agreed. @MdSahil-oss can you refactor so `handleCleanShutdown` (which really should be called `handleUncleanShutdown`) is also exported and called if the check or repair fail?

Reply:
@haircommander done 👍🏽